src/ucp/wireup/select.c

```diff
@@ -849,31 +853,30 @@ static int ucp_wireup_is_lane_proxy(ucp_ep_h ep, ucp_rsc_index_t rsc_index,
             UCT_IFACE_FLAG_EVENT_RECV_SIG);
 }
 
-static inline int ucp_wireup_is_am_required(ucp_ep_h ep,
-                                            const ucp_ep_params_t *params,
-                                            unsigned ep_init_flags,
-                                            ucp_wireup_lane_desc_t *lane_descs,
-                                            int num_lanes_p)
+static inline int
+ucp_wireup_is_am_required(const ucp_wireup_select_ctx_t *select_ctx)
 {
+    ucp_ep_h ep = select_ctx->ep;
     ucp_lane_index_t lane;
 
     /* Check if we need active messages from the configurations, for wireup.
      * If not, check if am is required due to p2p transports */
 
-    if ((ep_init_flags & UCP_EP_CREATE_AM_LANE) ||
-        (params->field_mask & UCP_EP_PARAM_FIELD_SOCK_ADDR)) {
+    if ((select_ctx->ep_init_flags & UCP_EP_CREATE_AM_LANE) ||
```

Comment:
why are we checking UCP_EP_CREATE_AM_LANE and not ->need_am?
need to remove the constant UCP_EP_CREATE_AM_LANE altogether..

Reply:
@yosefe there are a lot of places in UCP that set `UCP_EP_CREATE_AM_LANE` flag in `ep_init_flags` for calling `ucp_wireup_init_lanes`.
do we want to replace this code by calling `ucp_wireup_init_lanes` w/ `need_am`  flag set to `0`/`1` depending on the circumstances?

Reply:
so in this seems better to do as you suggested in https://github.com/openucx/ucx/pull/4071#discussion_r315276375 and remove need_am

Reply:
> so in this seems better to do as you suggested in [#4071 (comment)](https://github.com/openucx/ucx/pull/4071#discussion_r315276375) and remove need_am

it's already done
`need_am` removed and we assign `UCP_EP_CREATE_AM_LANE` flag to `select_ctx::ep_init_flags` from RMA/AMO selection logic if it's needed